hfile.c

```diff
@@ -573,6 +573,9 @@ static hFILE *hopen_fd(const char *filename, const char *mode)
 
 hFILE *hdopen(int fd, const char *mode)
 {
+#if defined HAVE_SETMODE && defined O_BINARY
+    if (setmode(fd, O_BINARY) < 0) return NULL;
+#endif
```

Comment:
Without this patch, the test cases in test_bgzf.c which use the function try_bgzf_dopen fail on Windows, because the files are opened in text mode, and then the file descriptors are passed into bgzf_dopen and hdopen.

I considered changing the test cases instead, but couldn't really see why it would be a bad thing to always set binary mode on the file descriptors. Are there any cases where it would be a problem?